src/sentry/snuba/entity_subscription.py

```diff
@@ -0,0 +1,274 @@
+import re
+from abc import ABC, abstractmethod
+from copy import copy
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Optional
+
+from sentry.constants import CRASH_RATE_ALERT_SESSION_COUNT_ALIAS
+from sentry.eventstore import Filter
+from sentry.exceptions import InvalidQuerySubscription, UnsupportedQuerySubscription
+from sentry.models import Environment
+from sentry.release_health.metrics import metric_id, tag_key
+from sentry.search.events.fields import resolve_field_list
+from sentry.search.events.filter import get_filter
+from sentry.snuba.dataset import EntityKey
+from sentry.snuba.models import QueryDatasets, SnubaQueryEventType
+from sentry.utils.snuba import Dataset, resolve_column, resolve_snuba_aliases
+
+# TODO: If we want to support security events here we'll need a way to
+# differentiate within the dataset. For now we can just assume all subscriptions
+# created within this dataset are just for errors.
+DATASET_CONDITIONS: Mapping[QueryDatasets, str] = {
+    QueryDatasets.EVENTS: "event.type:error",
+    QueryDatasets.TRANSACTIONS: "event.type:transaction",
+}
+ENTITY_TIME_COLUMNS: Mapping[str, str] = {
+    EntityKey.Events.value: "timestamp",
+    EntityKey.Sessions.value: "started",
+    EntityKey.Transactions.value: "finish_ts",
+    EntityKey.MetricsCounters.value: "timestamp",
+}
```

Comment:
Yeah here https://github.com/getsentry/snuba-sdk/blob/main/snuba_sdk/entity.py#L50
I kind of interpreted the comment above it that it might be removed one day, which is why I didn't reuse it
Not entirely sure though